lib/ansible/plugins/lookup/nios.py

```diff
@@ -108,7 +108,8 @@ def run(self, terms, variables=None, **kwargs):
         provider = kwargs.pop('provider', {})
         wapi = WapiLookup(provider)
         res = wapi.get_object(obj_type, filter_data, return_fields=return_fields)
-        for obj in res:
-            if 'extattrs' in obj:
-                obj['extattrs'] = flatten_extattrs(obj['extattrs'])
+        if res is not None:
+            for obj in res:
+                if 'extattrs' in obj:
+                    obj['extattrs'] = flatten_extattrs(obj['extattrs'])
         return res
```

Comment:
Yes, we should return empty list instead of None. Also, wanted to confirm that should I initialise the res with empty list or in else block return an empty list?

Reply:
Yes, empty list should be returned instead of None, will update the changes and checkin again